src/pymovements/measure/measures.py

```diff
@@ -50,11 +52,111 @@ def null_ratio(column: str, column_dtype: pl.DataType) -> pl.Expr:
     elif column_dtype == pl.List:
         non_null_lengths = pl.col(column).list.drop_nulls().drop_nans().list.len()
-        value = 1 - (non_null_lengths == pl.col(column).list.len()).sum() / pl.col(column).len()
+        value = (
+            1 - (non_null_lengths == pl.col(column).list.len()).sum() /
+            pl.col(column).len()
+        )
     else:
         raise TypeError(
             'column_dtype must be of type {Float64, Int64, Utf8, List}'
             f' but is of type {column_dtype}',
         )
 
     return value.alias('null_ratio')
+
+
+@register_sample_measure
+def data_loss(
+        timestamp_col: str = 'time',
+        *,
+        start_time: float | None = None,
+        end_time: float | None = None,
+        unit: Literal['count', 'time', 'ratio'] = 'count',
+) -> pl.Expr:
+    """Measure gaze data loss within a group using only Polars expressions.
+
+    Computes the number of missing samples ("count"), missing time ("time"),
+    or the ratio of missing to expected samples ("ratio"). The expected
+    sampling is derived from the group's median inter-sample interval (ISI)
+    computed on the ``timestamp_col`` after sorting.
+
+    If ``start_time``/``end_time`` are not provided, the group's first/last
+    timestamps are used (i.e., ``min``/``max`` of ``timestamp_col``).
+
+    Parameters
+    ----------
+    timestamp_col: str
+        Name of the timestamp column. Should be numeric and (ideally) monotonic
+        within the group.
+    start_time: float | None
+        Recording start time for the group. If ``None``, uses the group's first timestamp.
+    end_time: float | None
+        Recording end time for the group. If ``None``, uses the group's last timestamp.
+    unit: Literal['count', 'time', 'ratio']
+        - "count": number of missing samples (absolute)
+        - "time": missing time in timestamp units (approx. ``missing_count * median_ISI``)
+        - "ratio": ``missing_count / expected`` (in [0, 1])
+
+    Returns
+    -------
+    pl.Expr
+        A scalar (per-group) expression.
+
+    Raises
+    ------
+    ValueError
+        If ``unit`` is not one of 'count', 'time', or 'ratio'.
+    TypeError
+        If ``timestamp_col`` is not a string.
+    """
+    if not isinstance(timestamp_col, str):
+        raise TypeError(
+            f'timestamp_col must be a string, but got: {type(timestamp_col).__name__}',
+        )
+
+    timestamps = pl.col(timestamp_col)
+
+    # Determine group anchors: provided or derived.
+    start_expr = pl.lit(start_time) if start_time is not None else timestamps.min()
+    end_expr = pl.lit(end_time) if end_time is not None else timestamps.max()
+
+    # Observed samples in the group.
+    observed = pl.len()
+
+    # Median inter-sample interval (after sorting). Drop nulls before median.
+    isi_median = timestamps.sort().diff().drop_nulls().median()
+
+    # Expected sample count over [start, end] with inclusive endpoints.
```

Comment:
the data loss should not just be limited to the time column.

Let's take this samples data frame:

| time | pixel |
|------|-------|
| 1 | [111, 123] |
| 2 | [111, 123] |
| 3 | null |
| 4 | null |
| 5 | [111, 123] |
| 9 | [111, 123] |

There are two instances of data loss: one through invalid values (timesteps 3 to 4) and one through completely missing rows (timesteps 6 to 8).

Reply:
let's leave out the ` | None = None` and make both `column` and `time_column` a required argument.

that's more explicit and we have one less what-if to document and care about.